test/core/memory_usage/memory_usage_test.cc

```diff
@@ -72,26 +75,43 @@ int main(int argc, char** argv) {
   } else {
     strcpy(root, ".");
   }
-  /* start the server */
-  Subprocess svr({absl::StrCat(root, "/memory_usage_server",
-                               gpr_subprocess_binary_extension()),
-                  "--bind", grpc_core::JoinHostPort("::", port), "--nosecure",
-                  absl::StrCat("--minstack=", absl::GetFlag(FLAGS_minstack))});
-
-  /* start the client */
-  Subprocess cli({absl::StrCat(root, "/memory_usage_client",
-                               gpr_subprocess_binary_extension()),
-                  "--target", grpc_core::JoinHostPort("127.0.0.1", port),
-                  absl::StrCat("--warmup=", absl::GetFlag(FLAGS_warmup)),
-                  absl::StrCat("--benchmark=", absl::GetFlag(FLAGS_benchmark)),
-                  absl::StrCat("--minstack=", absl::GetFlag(FLAGS_minstack))});
-
-  /* wait for completion */
-  if ((status = cli.Join()) != 0) {
-    printf("client failed with: %d", status);
-    return 1;
+
+  /* Set configurations */
+  std::string minstack_arg = "--nominstack";
```

Comment:
I'm thinking of leave it the way it is because for client, only minstack is a flag so minstack_arg would still need to be declared

Reply:
I've got some grave concerns about what this code looks like once we have 10 scenarios... since someone's going to come through and write `if (GetFlag(...) == "scenario_a" || GetFlag(...) == "scenario_b") { }` and later someone else is going to add a stanza `if (GetFlag(...) == "scenario_b" || GetFlag(...) == "scenario_c") {}` and then the only way to understand the set of flags that will be set is to mentally execute hundreds of lines of code.

That said, that client and server have different argument lists is a completely valid concern... how about:
```
struct ScenarioArgs {
  std::vector<std::string> client;
  std::vector<std::string> server;
};
const std::map<std::string /*scenario*/, ScenarioArgs> scenarios = {
  {"secure", {/*client=*/{"--secure"}, /*server=*/{"--secure"}}},
  {"resource_quota", {/*client=*/{"--secure"}, /*server=*/{"--secure"}}},
  {"minstack", {/*client=*/{"--minstack"}, /*server=*/{}}},
  {"default", {{}, {}}},
};
auto it_scenario = scenarios.find(FLAGS_scenario_config);
if (it_scenario == scenarios.end()) { crash_with_usage(); }
const std::vector<std::string> client_flags_for_scenario = it_scenario->second.client;
const std::vector<std::string> server_flags_for_scenario = it_scenario->second.server;
```

Reply:
Understandable concern, I added in the changes and adjusted some things, how does it look? Also by crash_with_usage() did you mean something like return 2?